test/unit/src/common/components/repository-row/t_repository-row.js

```diff
@@ -1,90 +1,116 @@
+        expect(view.find(UnconfiguredDropdown).length).toBe(1);
+      });
+    });
+
+    context('registerNameIsOpen prop is true', function() {
+      it('should render dropdown', function() {
+        view.setProps({ registerNameIsOpen: true });
+        expect(view.find(RegisterNameDropdown).length).toBe(1);
+      });
+    });
+
+    context('toggleDropdownState', function() {
+      let component;
+
+      beforeEach(function() {
+        view = shallow(<RepositoryRowView { ...props }/>);
+        component = view.instance();
+      });
+
+      it('should render NameMismatchDropdown', function() {
+        component.toggleDropdownState('nameMismatchDropdownExpanded');
+        expect(view.find(NameMismatchDropdown).length).toBe(1);
+      });
+
+      it('should render UnconfiguredDropdown', function() {
+        component.toggleDropdownState('unconfiguredDropdownExpanded');
+        expect(view.find(UnconfiguredDropdown).length).toBe(1);
+      });
+
+      it('should render UnregisteredDropdown', function() {
+        component.toggleDropdownState('unregisteredDropdownExpanded');
+        expect(view.find(RegisterNameDropdown).length).toBe(1);
+      });
+
+      it('should render RemoveRepoDropdown', function() {
+        component.toggleDropdownState('removeDropdownExpanded');
+        expect(view.find(RemoveRepoDropdown).length).toBe(1);
+      });
+    });
   });
 
   context('when latest build log is available', () => {
     beforeEach(() => {
       // shallow render container component and get view from it
-      view = shallow(
-        <RepositoryRow
-          store={store}
-          snap={snap}
-          latestBuild={latestBuild}
-          fullName={fullName}
-          authStore={authStore}
-          registerNameStatus={registerNameStatus}
-        />)
-        .find('RepositoryRow').dive();
+      view = shallow(<RepositoryRowView { ...props }/>);
     });
 
     it('should render Row', () => {
       expect(view.type()).toEqual(Row);
     });
 
     it('should contain BuildStatus linked to build page', () => {
-      const expectedUrl = `/${fullName}/builds/${latestBuild.buildId}`;
+      const expectedUrl = `/${props.fullName}/builds/${props.latestBuild.buildId}`;
+
       expect(view.find('BuildStatus').length).toBe(1);
       expect(view.find('BuildStatus').prop('link')).toBe(expectedUrl);
     });
   });
 
   context('when latest build log is not yet available', () => {
     beforeEach(() => {
-      const buildWithoutLog = {
-        ...latestBuild,
-        buildLogUrl: null
-      };
-
-      // shallow render container component and get view from it
-      view = shallow(
-        <RepositoryRow
-          store={store}
-          snap={snap}
-          latestBuild={buildWithoutLog}
-          fullName={fullName}
-          authStore={authStore}
-          registerNameStatus={registerNameStatus}
-        />)
-        .find('RepositoryRow').dive();
+      Object.assign(props.latestBuild, { buildLogUrl: null });
```

Comment:
This overrides `props.latestBuild`, so it may affect other tests that use `props` outside of this context.